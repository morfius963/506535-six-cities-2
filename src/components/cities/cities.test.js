import React from "react";
import renderer from "react-test-renderer";
import Cities from "./cities.jsx";
import fixtureData from "../../__fixtures__/offers.js";

describe(`snapshot test`, () => {
  it(`Component correctly renders`, () => {
    const allCities = Array.from(new Set(fixtureData.map((offer) => offer.city.name)));
    const tree = renderer
      .create(
          <Cities
            cities = {allCities}
            activeCity={`Amsterdam`}
            onCityClick = {jest.fn()}
          />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
