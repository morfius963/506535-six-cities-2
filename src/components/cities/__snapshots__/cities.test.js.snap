// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test Component correctly renders 1`] = `
<section
  className="locations container"
  onClick={[Function]}
>
  <ul
    className="locations__list tabs__list"
  >
    <li
      className="locations__item"
    >
      <a
        className="locations__item-link tabs__item tabs__item--active"
        href="#"
      >
        <span>
          Amsterdam
        </span>
      </a>
    </li>
    <li
      className="locations__item"
    >
      <a
        className="locations__item-link tabs__item "
        href="#"
      >
        <span>
          Paris
        </span>
      </a>
    </li>
    <li
      className="locations__item"
    >
      <a
        className="locations__item-link tabs__item "
        href="#"
      >
        <span>
          Brussels
        </span>
      </a>
    </li>
    <li
      className="locations__item"
    >
      <a
        className="locations__item-link tabs__item "
        href="#"
      >
        <span>
          Hamburg
        </span>
      </a>
    </li>
    <li
      className="locations__item"
    >
      <a
        className="locations__item-link tabs__item "
        href="#"
      >
        <span>
          Dusseldorf
        </span>
      </a>
    </li>
  </ul>
</section>
`;
